pkg/agent/provisioning.go

```diff
@@ -0,0 +1,67 @@
+package agent
+
+import (
+	"github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	"github.com/Axway/agent-sdk/pkg/apic/provisioning"
+)
+
+// credential request definitions
+
+// createOrUpdateCredentialRequestDefinition -
+func createOrUpdateCredentialRequestDefinition(data *v1alpha1.CredentialRequestDefinition) (*v1alpha1.CredentialRequestDefinition, error) {
+	// TODO - check cache for credential request, update if needed
+	return agent.apicClient.RegisterCredentialRequestDefinition(data, true)
+}
+
+// NewCredentialRequestBuilder - called by the agents to build and register a new credential reqest definition
+func NewCredentialRequestBuilder() provisioning.CredentialRequestBuilder {
```

Comment:
Can we change the name to NewCredentialRequestDefinitionBuilder() or may be short NewCRDBuilder() to avoid having long name?

Reply:
same thing for the func above too

Reply:
done